File: frontend/types/customers.ts
export type CustomerWorkspaceRecentOrder = {
  sales_order_id: string;
  order_number: string;
  status: string;
  payment_status: string;
  total_amount: string;
  ordered_at: string | null;
};

export type CustomerWorkspaceRecentReturn = {
  sales_return_id: string;
  return_number: string;
  order_number: string;
  refund_status: string;
  refund_amount: string;
  requested_at: string | null;
};

export type CustomerWorkspaceItem = {
  customer_id: string;
  name: string;
  phone: string;
  email: string;
  address: string;
  total_orders: number;
  completed_orders: number;
  open_orders: number;
  total_returns: number;
  lifetime_revenue: string;
  outstanding_balance: string;
  last_order_at: string | null;
  last_return_at: string | null;
  recent_orders: CustomerWorkspaceRecentOrder[];
  recent_returns: CustomerWorkspaceRecentReturn[];
};

export type CustomerWorkspace = {
  query: string;
  items: CustomerWorkspaceItem[];
};

File: easy_ecom/api/schemas/customers.py
from decimal import Decimal

from pydantic import BaseModel


class CustomerWorkspaceRecentOrderResponse(BaseModel):
    sales_order_id: str
    order_number: str
    status: str
    payment_status: str
    total_amount: Decimal
    ordered_at: str | None


class CustomerWorkspaceRecentReturnResponse(BaseModel):
    sales_return_id: str
    return_number: str
    order_number: str
    refund_status: str
    refund_amount: Decimal
    requested_at: str | None


class CustomerWorkspaceItemResponse(BaseModel):
    customer_id: str
    name: str
    phone: str
    email: str
    address: str
    total_orders: int
    completed_orders: int
    open_orders: int
    total_returns: int
    lifetime_revenue: Decimal
    outstanding_balance: Decimal
    last_order_at: str | None
    last_return_at: str | None
    recent_orders: list[CustomerWorkspaceRecentOrderResponse]
    recent_returns: list[CustomerWorkspaceRecentReturnResponse]


class CustomerWorkspaceResponse(BaseModel):
    query: str = ''
    items: list[CustomerWorkspaceItemResponse]
